param/parameterized.py

```diff
@@ -974,7 +974,7 @@ class String(Parameter):
       class IPAddress(String):
         '''IPv4 address as a string (dotted decimal notation)'''
        def __init__(self, default="0.0.0.0", allow_None=False, **kwargs):
-           ip_regex = '^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$'
+           ip_regex = '^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$'
```

Comment:
```suggestion
           ip_regex = r'^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$'
```
?

Reply:
@mcepl, is the above change ok? I did it to match the rest of your changes, where you used raw strings rather than escaping, but I haven't tested that it's equivalent.

Reply:
Bit confusing, but if you expand the diff up a bit...

![Screenshot from 2020-09-17 22-53-19](https://user-images.githubusercontent.com/1929/93532294-a6ca9300-f938-11ea-8349-fa25e2b31af5.png)

I.e. we are well inside a string by now :)

There are two things:
1. suggesting the right way for someone to write this code themselves
2. not getting a deprecation warning (?) from param itself

The suggested edit would be ok for 1, but I guess it does not help for 2.

The original is presumably ok for 2 (that is what you were doing I guess?), and I _think_ it's ok for 1 also. But I can't remember exactly.